src/name_type.rs

```diff
@@ -59,14 +59,13 @@ macro_rules! convert_to_array {
     }};
 }
 
-/// NameType struct
-///
-/// NameType Struct can be created using the new function by passing, id as its parameter.
+/// NameType can be created using the new function by passing id as its parameter.
+#[derive(Default, Eq, PartialOrd, Ord)]
 pub struct NameType(pub [u8; NAME_TYPE_LEN]);
```

Comment:
This will help
https://github.com/rust-lang/rust/blob/master/src/libcore/array.rs